homeassistant/components/systemmonitor/sensor.py

```diff
@@ -408,46 +542,24 @@ async def async_setup_entry(  # noqa: C901
     loaded_resources: set[str] = set()
     disk_arguments = await hass.async_add_executor_job(get_all_disk_mounts)
     network_arguments = await hass.async_add_executor_job(get_all_network_interfaces)
-    cpu_temperature = await hass.async_add_executor_job(read_cpu_temperature)
+    temps = await hass.async_add_executor_job(psutil.sensors_temperatures)
+    cpu_temperature = await hass.async_add_executor_job(read_cpu_temperature, temps)
```

Comment:
Maybe put all these calls in a single function that we schedule once on the executor, and return a dict or a dataclass with the result eg?